src/hotspot/share/cds/classListParser.cpp

```diff
@@ -166,14 +170,17 @@ int ClassListParser::parse(TRAPS) {
 
 bool ClassListParser::parse_one_line() {
   for (;;) {
-    if (fgets(_line, sizeof(_line), _file) == nullptr) {
+    _line = _line_reader.read_line();
+    if (_line == nullptr) {
+      if (_line_reader.is_oom()) {
+        // Don't try to print the input line that we already know is too long.
+        _token = nullptr;
+        _line_len = 0;
+        error("Input line too long"); // will exit JVM
```

Comment:
The line number is printed inside `error()`.

Also, I changed the error message to "Out of memory", as we'd come here only if LineReader fails to expand its internal buffer. If the line is too long, it will be broken into multiple chunks. See the gtest case.